### ISSUE TITLE
us.i.posthog.com blocked by corporate filter

### ISSUE THREAD

**Issue by marking** [Aider-AI/aider#2532]
### Issue

For whatever reason, my company blocks access to us.i.posthog.com. This results in many errors such as the following:
```
error uploading: HTTPSConnectionPool(host='us.i.posthog.com', port=443): Max retries exceeded with url: /batch/ (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))
```

I'm wondering if there is a way to either disable telemetry or at least suppress the errors from being printed to stderr/stdout. I looked in the `.aider.conf.yml` I have stored in my $HOME, but I didn't see anything related to 'tracking', 'telemetry', or 'usage'.

### Version and model info

```
Aider v0.66.0
Main model: bedrock/anthropic.claude-3-5-sonnet-20241022-v2:0 with architect edit format, prompt 
cache, infinite output
Editor model: bedrock/anthropic.claude-3-5-sonnet-20241022-v2:0 with editor-diff edit format     
Weak model: bedrock/us.anthropic.claude-3-5-haiku-20241022-v1:0
Git repo: .git with 0 files
Repo-map: using 1024 tokens, files refresh
```

**Comment by marking** [comment]
By looking at the code in aider/analytics.py, I was able to find the $HOME/.aider/analytics.json file and edit it to set
```
"permanently_disable": true
```
I must admit, the question to opt into telemetry blew by me so quickly (along with the error messages) that I barely noticed it. I still think it would be nice to have a command line arg and/or .yml file setting to disable telemetry or at least ask each time.

**Comment by paul-gauthier** [comment]
Thanks for trying aider and filing this issue.

This doc may be helpful, as it discusses all the settings for analytics:

https://aider.chat/docs/more/analytics.html

Once you've opted-in, analytics are on by default. You can disable analytics per-session with `--no-analytics`. Or you can run aider with `--analytics-disable` to turn them off forever. Your edit to the json file has also turned it off forever.

I've updated aider so that if posthog has a connection error like this, it will be disabled for the rest of the session.

The change is available in the main branch. You can get it by installing the latest version from github:

```
aider --install-main-branch

# or...

python -m pip install --upgrade --upgrade-strategy only-if-needed git+https://github.com/Aider-AI/aider.git
```

If you have a chance to try it, let me know if it works better for you.

**Comment by paul-gauthier** [comment]
I'm going to close this issue for now, but feel free to add a comment here and I will re-open. Or feel free to file a new issue any time.
